docs/historical_model.rst

```diff
@@ -122,13 +122,42 @@ Users can specify a custom model name via the constructor on
 ``HistoricalRecords``. The common use case for this is avoiding naming conflict
 if the user already defined a model named as 'Historical' + model name.
 
+This feature provides the ability to override the default model name used for the generated history model.
+
+To configure history models to use a different name for the history model class, use an option named ``custom_model_name``.
+The value for this option can be a `string` or a `callable`.
+A simple string replaces the default name of `'Historical' + model name` with the defined string.
+The most simple use case is illustrated below using a simple string:
+
 .. code-block:: python
 
     class ModelNameExample(models.Model):
         history = HistoricalRecords(
             custom_model_name='SimpleHistoricalModelNameExample'
         )
 
+If you are using a base class for your models and want to apply a name change for the historical model
+for all models using the base class then a callable can be used.
+The callable is passed the name of the model for which the history model will be created.
+As an example using the callable mechanism, the below changes the default prefix `Historical` to `Audit`:
+
+.. code-block:: python
+
+    class Poll(models.Model):
+        question = models.CharField(max_length=200)
+        history = HistoricalRecords(custom_model_name=lambda x:f'Audit{x}')
+
+    class Opinion(models.Model):
+        opinion = models.CharField(max_length=2000)
+
+    register(Opinion, custom_model_name=lambda x:f'Audit{x}')
+    
+The resulting history class names would be `AuditPoll` and `AuditOpinion`.
+If the app the models are defined in is `yoda` then the corresponding history table names would be `yoda_auditpoll` and `yoda_auditopinion`
```

Comment:
Is this true? looking at `table_name` param and trying to figure out how this might conflict with it. What happens if I specify `table_name` and `custom_model_name`?

Reply:
Specifying the table_name param overrides any custom_model_name setting since it explicitly sets the db_table meta attrib (https://github.com/treyhunner/django-simple-history/blob/372f21268963cd52f25e3d7915c48dd77c468d5a/simple_history/models.py#L218)

So the above is true only if the table_name param is not set.
The tests written for this feature verify both class name and table name change as a result of using custom_model_name ( https://github.com/treyhunner/django-simple-history/pull/490/files#diff-0772dbe490161136d6466f53ba7c2301R754)